libr/anal/type.c

```diff
@@ -330,35 +279,81 @@ static RAnalBaseType *get_union_type(RAnal *anal, const char *sname) {
 	return NULL;
 }
 
+static RAnalBaseType *get_typedef_type(RAnal *anal, const char *sname) {
+	r_return_val_if_fail (anal && sname, NULL);
+
+	RAnalBaseType *base_type = r_anal_base_type_new (R_ANAL_BASE_TYPE_KIND_TYPEDEF);
+	if (!base_type) {
+		return NULL;
+	}
+
+	base_type->type = get_type_data (anal->sdb_types, "typedef", sname);
+	if (!base_type->type) {
+		goto error;
+	}
+	return base_type;
+
+error:
+	r_anal_base_type_free (base_type);
+	return NULL;
+}
+
+static RAnalBaseType *get_atomic_type(RAnal *anal, const char *sname) {
+	r_return_val_if_fail (anal && sname, NULL);
+
+	RAnalBaseType *base_type = r_anal_base_type_new (R_ANAL_BASE_TYPE_KIND_ATOMIC);
+	if (!base_type) {
+		return NULL;
+	}
+
+	base_type->type = get_type_data (anal->sdb_types, "type", sname);
+	if (!base_type->type) {
+		goto error;
+	}
+
+	RStrBuf key;
+	r_strbuf_init (&key);
+	r_strbuf_setf (&key, "type.%s.size", sname);
```

Comment:
whats the point on using strbuf here?

Reply:
It's slightly more efficient than r_str_newf() when the string fits in the small stack buffer.

Reply:
What about having an initf vonstructor with format?

Reply:
Sounds like a good idea to me, but out of scope here.

Reply:
done: https://github.com/radareorg/radare2/pull/17634

Reply:
yeah, please update this pr when the initf gets merged, it lgtm, waiting for @ret2libc to review it